homeassistant/components/icloud/account.py

```diff
@@ -105,12 +106,22 @@ def setup(self) -> None:
             return
 
         user_info = None
+        api_devices = {}
         try:
+            api_devices = self.api.devices
             # Gets device owners infos
-            user_info = self.api.devices.response["userInfo"]
-        except PyiCloudNoDevicesException:
+            user_info = api_devices.response["userInfo"]
```

Comment:
if `api_devices` is a dict, how can you call `api_devices.response` ?

Reply:
That's what I thought at first glance, but while testing, it is actually working.

![Uploading Capture d’écran 2020-03-04 à 23.39.03.png…]()

Reply:
The devices property is returning a "service" with devices as attr.


https://github.com/picklepete/pyicloud/blob/7ff594e2c8abdd5e0f5990957597c8eead63d33a/pyicloud/base.py#L302-L308


https://github.com/picklepete/pyicloud/blob/7ff594e2c8abdd5e0f5990957597c8eead63d33a/pyicloud/services/findmyiphone.py#L77-L78

Reply:
PS : That will change with my refactor of the service picklepete/pyicloud#227